src/lib/viewers/text/CSVViewer.js

```diff
@@ -53,43 +52,53 @@ class CSVViewer extends TextBaseViewer {
-        const assetUrlCreator = createAssetUrlCreator(location);
-        const papaWorkerUrl = assetUrlCreator(JS[0]);
 
         return Promise.all([this.loadAssets(JS), this.getRepStatus().getPromise()])
             .then(() => {
-                this.api.get(papaWorkerUrl, { type: 'blob' }).then(papaWorkerBlob => {
-                    /* global Papa */
-                    const workerSrc = URL.createObjectURL(papaWorkerBlob);
-                    Papa.SCRIPT_PATH = workerSrc;
-
-                    this.startLoadTimer();
-                    const urlWithAuth = this.createContentUrlWithAuthParams(template);
-                    Papa.parse(urlWithAuth, {
-                        download: true,
-                        error: (err, file, inputElem, reason) => {
-                            const error = new PreviewError(ERROR_CODE.LOAD_CSV, __('error_refresh'), { reason });
-                            this.handleDownloadError(error, urlWithAuth);
-                        },
-                        complete: results => {
-                            if (this.isDestroyed() || !results) {
-                                return;
-                            }
-
-                            this.checkForParseErrors(results);
-
-                            this.data = results.data;
-                            this.finishLoading();
-                            URL.revokeObjectURL(workerSrc);
-                        },
-                    });
+                this.startLoadTimer();
+                const urlWithAuth = this.createContentUrlWithAuthParams(template);
+                Papa.parse(urlWithAuth, {
+                    delimiter: this.getDelimiter(extension),
+                    download: true,
+                    error: (err, file, inputElem, reason) => {
+                        const error = new PreviewError(ERROR_CODE.LOAD_CSV, __('error_refresh'), { reason });
+                        this.handleDownloadError(error, urlWithAuth);
+                    },
+                    complete: results => {
+                        if (this.isDestroyed() || !results) {
+                            return;
+                        }
+
+                        this.checkForParseErrors(results);
+
+                        this.data = results.data;
+                        this.finishLoading();
+                    },
+                    worker: true,
                 });
             })
             .catch(this.handleAssetError);
     }
 
+    getDelimiter(extension) {
+        // Papaparse will by default autodetect the delimiter but in cases where the content has many
+        // special characters, it can get confused. Since we have the file extension, rely on that for
+        // the explicit delimiter, otherwise default to the autodetection.
+        switch (extension) {
+            case 'csv':
+                return ',';
+            case 'tsv':
+                return '\t';
+            default:
+                return '';
```

Comment:
Does returning empty string fall back to auto-detection? Since it's CSV (Comma-separated values) viewer, should the default be comma?

Reply:
Yeah, empty string falls back to auto-detection. Since we currently rely on auto-detection, I figured we would want to keep that default (even though we only support csv and tsv currently)